### PR TITLE
Implement multiple traits for all objects

### PR DESCRIPTION
**Pull request by vitvakatu**

### CODE REVIEW

File: src/animation.rs [hunk 1]
```diff
@@ -234,6 +236,26 @@ pub struct Action {
...
+}
+
+impl Eq for Action {}
+
+impl Hash for Action {
+    fn hash<H: Hasher>(
+        &self,
+        state: &mut H,
+    ) {
+        self.pointer.hash(state);
```
Comment (alteous) [inline src/animation.rs]: Is there any way the value `self.pointer` can change? If not, then all is well.

File: src/object.rs [hunk 2]
```diff
@@ -84,7 +105,7 @@ impl Object {
     }
 
     /// Add new [`Object`](struct.Object.html) to the group.
-    pub fn set_parent<P: Deref<Target=Object>>(
+    pub fn set_parent<P: Deref<Target = Object>>(
         &mut self,
```
Comment (alteous) [inline src/object.rs]: Is `P: AsRef<Object>` achievable instead?

Comment (vitvakatu) [reply src/object.rs]: Checking it now...

File: src/animation.rs (hunk 1 above)
Comment (vitvakatu) [reply src/animation.rs]: I don't think it's possible

File: src/object.rs (hunk 2 above)
Comment (vitvakatu) [reply src/object.rs]: Replaced, check it out

**Pull request merged**